MongoEngine + Typeform REST changes

jay-khatri opened the pull request:
General Changes:

1. Added parsing of a request to Typeform's data API
2. Give's the user a "preview" of the schema before uploading
2. Treating email as an identifier, so any repeats are discarded when uploading data
4. Message text div added to some templates for easier debugging
5. Switched to use of MongoEngine

jay-khatri committed:
finished csv parsing, not finished with html for full feature

jay-khatri committed:
updated parser with rest request, and made a conversion from mongodb to mongoEngine.

jay-khatri committed:
changed the parsing algorithms slightly, with recent suggestions

jay-khatri committed:
added a few updates to the manual uploading/deleting of an entry

jay-khatri committed:
added a few updates to the manual uploading/deleting of an entry

jay-khatri committed:
fixed merging and added save_group method for readability

him229 commented on resources/typeform.py:
```diff
@@ -0,0 +1,106 @@
+            try:
+                # if the person object exists with a given email, ignore it
+                self.person_db.objects.get(email=person_data["email"])
+                print person_data["email"]
```
This print should just be for debugging purposes since it wont be displayed on the template

him229 commented on resources/typeform.py:
```diff
@@ -0,0 +1,106 @@
+            responses = requests.get(self.request_string +"&offset=" + str(request_range * DEFAULT_REQUEST_LIMIT)).json()["responses"]
+            #if the number of entries is less than 1000
+            if responses:
+                for x in responses:
```
what's `x`? Just as a habit I would give it a meaningful name like `for response in responses`

him229 commented on resources/typeform.py:
```diff
@@ -0,0 +1,106 @@
+                    if entry:
+                        # create a new dictionary
+                        curr_dic = {}
+                        for x,y in entry.iteritems():
```
A good practise is to write it like this `for k,v in entry.iteritems()` where k is the key and v is the value

him229 reviewed (commented):
Looks good. Thanks for accommodating all changes we spoke about. Just some minor edits and you should be good to merge. Lastly, I haven't run it locally so please MAKE SURE this works 👍

jay-khatri committed:
added new class for handling saving/deleting of (it'll come in handy in the future)

added new class for handling saving/deleting of (it'll come in handy in the future)

jay-khatri committed:
added new class for handling saving/deleting of (it'll come in handy in the future)

jay-khatri committed:
added new class for handling saving/deleting of (it'll come in handy in the future)

jay-khatri committed:
typeform test functionality added, checks for output correct output contents, count, etc.

jay-khatri committed:
app running/working with full functionality, still need to add test for mongo_engine, but app-related functionality can be built on

jay-khatri committed:
tests passed and merge conflicts resolved for one unit test

jay-khatri committed:
mongo_interface_test working and finalized changes on latest merge; next step is to work on second test
